.travis.yml

```diff
@@ -0,0 +1,36 @@
+language: node_js
+sudo: false
+node_js:
+  - '6'
+
+cache:
+  yarn: true
+
+jobs:
+  include:
+    - stage: test
+      addons:
+        sauce_connect:
+          username: angular-ci
+        jwt:
+          secure: "NJ6eWX3YarBTf2LOjOXFhGfRQ90WQ4ZrukAHUxTyUc1o3AXURb3WRi1HzXFh4fnlliBuaCY85/B5oSctvVg+evAOfnqOPm0WN/AP4/PIY6Xicc3jJDmyaeHtfvq2hb6QwxOciM/Ht7WxPsmlIR0wLz98CwqrOPxltjj25E9pPK4="
+      script:
+        - mkdir build
+        - travis_retry yarn run build copy test
+    - stage: deploy
+      if: type != pull_request AND branch = master
```

Comment:
Where do `type` and `branch` come from?

Reply:
That's a special travis "language": https://docs.travis-ci.com/user/conditional-builds-stages-jobs#Specifying-conditions